File: app/src/main/java/aromko/de/wishlist/model/ChatMessage.java
package aromko.de.wishlist.model;

import com.google.firebase.database.Exclude;

public class ChatMessage {
    @Exclude
    private String wishId;
    private String name;
    private String text;
    private long timestamp;

    public ChatMessage() {
    }

    public ChatMessage(String name, String text, long timestamp) {
        this.name = name;
        this.text = text;
        this.timestamp = timestamp;
    }

    public String getWishId() {
        return wishId;
    }

    public void setWishId(String wishId) {
        this.wishId = wishId;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public long getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(long timestamp) {
        this.timestamp = timestamp;
    }

    @Override
    public String toString() {
        return "ChatMessage{" +
                "wishId='" + wishId + '\'' +
                ", name='" + name + '\'' +
                ", text='" + text + '\'' +
                ", timestamp=" + timestamp +
                '}';
    }
}

File: app/src/main/java/aromko/de/wishlist/model/ChatMessage.kt
package aromko.de.wishlist.model

import com.google.firebase.database.Exclude

class ChatMessage {
    @Exclude
    var wishId: String? = null
    var name: String? = null
    var text: String? = null
    var timestamp: Long = 0
    var userId: String? = null
    var allowedUsers: MutableMap<String?, Any?>? = null
    var wishName: String? = null
    var wishListId: String? = null

    constructor()
    constructor(
        name: String?,
        text: String?,
        timestamp: Long,
        userId: String?,
        allowedUsers: MutableMap<String?, Any?>?,
        wishName: String?,
        wishListId: String?
    ) {
        this.name = name
        this.text = text
        this.timestamp = timestamp
        this.userId = userId
        this.allowedUsers = allowedUsers
        this.wishName = wishName
        this.wishListId = wishListId
    }

    override fun toString(): String {
        return "ChatMessage(wishId=$wishId, name=$name, text=$text, timestamp=$timestamp, userId=$userId, allowedUsers=$allowedUsers, wishName=$wishName, wishListId=$wishListId)"
    }
}